api4/system.go

```diff
@@ -419,15 +420,18 @@ func getAnalytics(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getSupportedTimezones(c *Context, w http.ResponseWriter, r *http.Request) {
-	supportedTimezones := c.App.Timezones()
+	supportedTimezones := c.App.Timezones.GetSupported()
+	if supportedTimezones == nil {
+		supportedTimezones = make([]string, 0)
+	}
 
-	if supportedTimezones != nil {
-		w.Write([]byte(model.TimezonesToJson(supportedTimezones)))
-		return
+	b, err := json.Marshal(supportedTimezones)
+	if err != nil {
+		c.Log.Warn("Unable to marshal JSON in timezones.", mlog.Err(err))
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
I noticed in Martin's LDAP PR he was using this pattern for JSON serialization too. Is this the new "best practice" for JSON serialization in the `api4` package now? Can we document it in some "best practices" guide somewhere?

Reply:
Yes this is the best practice now. I think this is in the dev decisions doc.